add --extra-gen-snapshot-options support for build_aar command.

00SunnyDay00 opened the pull request:
Tiny patch for https://github.com/flutter/flutter/pull/69592

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

00SunnyDay00 committed:
add --extra-gen-snapshot-options support for build_aar command.

00SunnyDay00 commented:
@googlebot I signed it!

jmagman reviewed (changes requested):
This needs tests.  See existing `build_apk_test` for example.
https://github.com/flutter/flutter/blob/9e88fe328efcf42d3037a965b39269f0805cd4ce/packages/flutter_tools/test/commands.shard/permeable/build_apk_test.dart#L213

00SunnyDay00 committed:
add test for ExtraDartFlagOptions support

00SunnyDay00 committed:
resolve conflicts

# Conflicts:
#	packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart

00SunnyDay00 commented:
> This needs tests. See existing `build_apk_test` for example.
> 
> https://github.com/flutter/flutter/blob/9e88fe328efcf42d3037a965b39269f0805cd4ce/packages/flutter_tools/test/commands.shard/permeable/build_apk_test.dart#L213



> This needs tests. See existing `build_apk_test` for example.
> 
> https://github.com/flutter/flutter/blob/9e88fe328efcf42d3037a965b39269f0805cd4ce/packages/flutter_tools/test/commands.shard/permeable/build_apk_test.dart#L213

done

00SunnyDay00 commented:
done

jmagman commented on packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart:
```diff
@@ -219,6 +224,28 @@ void main() {
+      globals.fs.directory(globals.fs.path.join(tempDir.path, 'flutter_project',
+          'build', 'host','outputs','repo')).createSync(recursive: true);
+
+      await runBuildAarCommand(projectPath,
```
This isn't testing that the options are used.  Take another look at how `build_apk` is validating the process being called. https://github.com/flutter/flutter/blob/9e88fe328efcf42d3037a965b39269f0805cd4ce/packages/flutter_tools/test/commands.shard/permeable/build_apk_test.dart#L213

00SunnyDay00 replied:
done

00SunnyDay00 replied:
done

00SunnyDay00 committed:
add more test case for ExtraDartFlagOptions support

00SunnyDay00 committed:
Fix the ci problem on windows

00SunnyDay00 committed:
Fix another ci problem on windows

00SunnyDay00 committed:
Remove useless code

jmagman commented on packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart:
```diff
@@ -219,6 +232,47 @@ void main() {
+        '--extra-front-end-options=bar',
+      ]), throwsToolExit(message: 'Gradle task assembleAarRelease failed with exit code 1'));
+      expect(processManager, hasNoRemainingExpectations);
+    },
+        overrides: <Type, Generator>{
+          AndroidSdk: () => mockAndroidSdk,
+          FlutterProjectFactory: () => FakeFlutterProjectFactory(tempDir),
+          ProcessManager: () => processManager,
+          FeatureFlags: () => TestFeatureFlags(isIOSEnabled: false),
+          AndroidStudio: () => FakeAndroidStudio(),
+        });
```
```suggestion
    },
    overrides: <Type, Generator>{
      AndroidSdk: () => mockAndroidSdk,
      FlutterProjectFactory: () => FakeFlutterProjectFactory(tempDir),
      ProcessManager: () => processManager,
      FeatureFlags: () => TestFeatureFlags(isIOSEnabled: false),
      AndroidStudio: () => FakeAndroidStudio(),
    });
```

00SunnyDay00 replied:
done

jmagman commented on packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart:
```diff
@@ -219,6 +232,47 @@ void main() {
+          '-Ptrack-widget-creation=true',
+          '-Ptree-shake-icons=true',
+          '-Ptarget-platform=android-arm,android-arm64,android-x64',
+          'assembleAarRelease'
```
```suggestion
          'assembleAarRelease',
```

00SunnyDay00 replied:
done

jmagman commented on packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart:
```diff
@@ -259,3 +313,15 @@ class FakeAndroidBuilder extends Fake implements AndroidBuilder {
   }
 }
+
+class FakeAndroidSdk extends Fake implements AndroidSdk {
+  FakeAndroidSdk(this.directory);
+
+  @override
+  final Directory directory;
+}
```
Overriding `directory` isn't necessary.
```suggestion
class FakeAndroidSdk extends Fake implements AndroidSdk { }
```

00SunnyDay00 replied:
done

jmagman commented on packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart:
```diff
@@ -190,9 +194,18 @@ void main() {
 
     setUp(() {
       tempDir = globals.fs.systemTempDirectory.createTempSync('flutter_tools_packages_test.');
+      mockAndroidSdk = FakeAndroidSdk(globals.fs.directory('irrelevant'));
```
Overriding `directory` isn't necessary.
```suggestion
      mockAndroidSdk = FakeAndroidSdk();
```

00SunnyDay00 replied:
done

jmagman reviewed (commented):
Some nits.
@blasten anything to add?

00SunnyDay00 committed:
fix CR issues

jmagman reviewed (approved):
LGTM awaiting review from @blasten to merge.

blasten reviewed (approved):
LGTM
